src/main/cpp/src/hive_hash.cu

```diff
@@ -203,18 +203,64 @@ class hive_device_row_hasher {
     {
     }
 
-    template <typename T, CUDF_ENABLE_IF(not cudf::is_nested<T>())>
-    __device__ hive_hash_value_t operator()(cudf::column_device_view const& col,
-                                            cudf::size_type row_index) const noexcept
-    {
-      return this->hash_functor.template operator()<T>(col, row_index);
-    }
+    struct stack_element {
+      cudf::column_device_view col;
+      int child_index;
+      hive_hash_value_t hash;
+
+      __device__ stack_element() = delete;
+      __device__ stack_element(cudf::column_device_view col) : col(col), child_index(0), hash(HIVE_INIT_HASH) {}
+    };
 
-    template <typename T, CUDF_ENABLE_IF(cudf::is_nested<T>())>
+    template <typename T>
     __device__ hive_hash_value_t operator()(cudf::column_device_view const& col,
                                             cudf::size_type row_index) const noexcept
     {
-      CUDF_UNREACHABLE("Nested type is not supported");
+      cudf::column_device_view curr_col = col.slice(row_index, 1);
+      stack_element stack[8] = {stack_element(curr_col), stack_element(curr_col), stack_element(curr_col), stack_element(curr_col),
```

Comment:
Why is the size "8" ?

Reply:
i can not understand how stack way works. Better to add some comment here.

Reply:
> Why is the size "8" ?

It's just a temporary solution, and I will improve it later.

Reply:
Move the 8 into the template.